demo/sb-container.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import { ScatterBasic } from '../src/visualizations'
+import { GlyphDot } from '@vx/glyph'
+
+const data = [{'label': 'Education', 'value_total': 0}, {'label': 'Visual and Performing Arts', 'value_total': 0}, {'label': 'Humanities', 'value_total': 0}, {'label': 'Sociology and Law', 'value_total': 15}, {'label': 'Business and PA', 'value_total': 10}, {'label': 'Phys Ed', 'value_total': 0}, {'label': 'Math, Comp and InfoSci', 'value_total': 0}, {'label': 'Engineering', 'value_total': 10}, {'label': 'Agriculture, NatRes', 'value_total': 0}, {'label': 'Health', 'value_total': 25}, {'label': 'Security', 'value_total': 0}, {'label': 'Other', 'value_total': 0}]
+
+const scatterData = data.map((d) => {
+  return {
+    fieldName: d.label,
+    value: d.value_total,
+  }
+})
+
+
+class ScatterBasicContainer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      fill: null,
+      radius: 4,
+      view: 'overview',
+    }
+  }
+
+  onPointHoverEnter = () => {
+    this.setState({
+      fill: '#b3ffff',
+      radius: (this.state.radius || 0) + 2,
+    })
+  }
+
+  onPointHoverLeave = () => {
+    this.setState({
+      fill: null,
+      radius: 4,
+    })
+  }
+
+  onPointClick = () => {
+    /**
+     * bring up modal for pie chart breakdown
+     */
+    this.setState({
+      view: 'detail',
+    })
+  }
+
+  render() {
+    /** draw a scattered point */
+    const drawAScatter = (left, top) => {
+      return (
+        <GlyphDot
+          key={Math.random()}
```

Comment:
So this is more commentary, and it's more intended for @woozyking. Please don't see this as a request for change at this time. 

The reason why math.random() as a key is a bad idea is that it causes issues with the process of reconciliation, where React tries to determine whether a subset of a component should be updated, or whether an entire component should be rendered. 

The end of the page I've linked to seems to suggest that when employing math.random() as a key, there is a much higher chance that the element in question will be rebuilt completely, rather than partially. 

If that's what is intended the context of this code, I'm not sure. d3 and charts being what they are, maybe stripping down and rebuilding is exactly what we want here. 

https://reactjs.org/docs/reconciliation.html

Reply:
I think both @y629wang and yourself have mixed this with the case from the last PR https://github.com/EQWorks/vz/pull/4#discussion_r215423506, where it did make sense and the intent was to re-render the tooltip to take advantage of the dynamically computed boundary offered by vx.

From the look of this case, there no reason to use `Math.random()` as key.